sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4900,6 +4959,8 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
   if (IndirectAccessTrackingEnabled)
     Queue->KernelsToBeSubmitted.push_back(Kernel);
 
+  if (auto Res = Queue->resetLastEventIfNeeded(CommandList))
+    return Res;
```

Comment:
Why is reset of the unused events can't happen in the executeCommandList?

Reply:
We probably can, but then we should add a new parameter to `executeCommandList` to separate the cases when `resetLastEventIfNeeded` is executed and when it is not. 
For example, one of the problematic cases: we should not reset the L0 event when closing the command-list if we take this event later as a dependency (this situation occurs when the event is from another queue, this happens inside `createAndRetainPiZeEventList` for `LastCommandEvent` which should be used as a dependency.)

I'm trying to say that the general rule is to reset the L0 event immediately after queue operations such as `zeCommandListAppendLaunchKernel`, `zeCommandListAppendMemoryCopy ` etc. (or after `zeCommandListAppendWaitOnEvents` if the queue operation used it, see next paragraph about it).

On the other hand, in the future we may do some optimization: currently for cases when we already had `zeCommandListAppendWaitOnEvents`, we could skip it in `resetLastEventIfNeeded` and for this we can add one parameter to `resetLastEventIfNeeded ` function, but if we transfer it to `executeCommandList`, then for such optimization you have to add two parameters to `executeCommandList` function. That is, we can do as you suggest, but the only question is whether it is necessary?